crates/subspace-service/src/sync_from_dsn.rs

```diff
@@ -212,30 +241,40 @@ async fn create_substrate_network_observer<Block>(
 }
 
 #[allow(clippy::too_many_arguments)]
-async fn create_worker<Block, AS, IQS, Client, PG>(
+async fn create_worker<Block, AS, IQS, Client, PG, B>(
     segment_headers_store: SegmentHeadersStore<AS>,
     node: &Node,
-    client: &Client,
-    import_queue_service: &mut IQS,
+    client: Arc<Client>,
+    import_queue_service: Arc<Mutex<Box<IQS>>>,
     sync_target_block_number: Arc<AtomicU32>,
     pause_sync: Arc<AtomicBool>,
     mut notifications: mpsc::Receiver<NotificationReason>,
     piece_getter: &PG,
+    network_service: Arc<NetworkService<Block, <Block as BlockT>::Hash>>,
+    mut notifications_sender: mpsc::Sender<NotificationReason>,
+    fast_sync_enabled: bool,
+    fast_sync_state: Arc<parking_lot::Mutex<Option<NumberFor<Block>>>>,
+    sync_service: Arc<SyncingService<Block>>,
 ) -> Result<(), sc_service::Error>
 where
+    B: sc_client_api::Backend<Block>,
```

Comment:
If `B` is first here, it should logically be first in the list of function generics as well, it is the case in all other cases here and elsewhere